### ISSUE TITLE
Hide Key Metrics banner/widget area when SC and GA4 are in a Gathering Data State

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#6607]
## Feature Description

This issue was originally part of #6209. However, the new "pre-loaded" gathering data state is being implemented in #5933. Once #5933 is merged, we can use the pre-loaded state to hide the Key Metrics banner/widget area, without causing the widget to "flicker".

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* If Search Console and / or GA4 are in a gathering data state:
    * The Key Metrics Setup CTA banner on the Site Kit dashboard should be hidden. 
    * The new Key Metrics Admin settings panel should be hidden.
* If Search Console and / or GA4 are not in a gathering data state, the Key Metrics Setup CTA and Admin Settings panel should be rendered. 
* When rendering or hiding the above components, there should be no "flicker" or movement on the page.

## Implementation Brief

In `assets/js/components/notifications/UserInputSettings.js`:

* Check if Search Console and Analytics-4 are in a gathering data state using the `isGatheringData` selector, respectively.
* If either is unresolved (i.e., `undefined`), return `null` to hide the component.
* If either is in a gathering data state, return `null` to hide the component.

### Test Coverage

* No tests are required for this change.

## QA Brief

* Enable the `userInput` feature flag.
* Set up Site Kit on a site without answering User Input Questions. Set up both Search Console and Analytics (GA4).
* Ensure the site is in a "Gathering data" state, either from being new or using the Tester plugin (for the Search Console - forcing GA4 report data is not yet supported by the Tester plugin).
* Dismiss all dashboard notifications
* Refresh the dashboard and note that the User Input questionnaire doesn't appear briefly then disappear (eg. "flash") when the page is loaded. It should never appear while in a "Gathering data" state.
* Similarly, ensure the Key Metrics section on the **Settings > Admin** page only appears when SC/GA4 are connected and _not_ in the "Gathering data" state.

## Changelog entry

* Hide Key Metric Widget area when Search Console and Google Analytics are still gathering data.

**Comment by bethanylang** [comment]
@jimmymadon Should this be added to the Key Metrics Widget epic? Can you also please add priority and Type labels here? Thanks!

**Comment by tofumatt** [comment]
ACs here are good. 👍🏻

**Comment by tofumatt** [comment]
IB ✅

**Comment by tofumatt** [comment]
Just a noted here: this has a failing E2E test that I haven’t been able to debug as my local environment has been timing out even on `develop`, but the PR has actually been approved and is otherwise ready-to-merge.

I need extra time to figure out what’s busted in my environment so I will unassign myself for now, @10upsimon has kindly offered to check out the tests here 🙂

**Comment by techanvil** [comment]
Please note, if this needs to go back to Execution from CR or QA, please assign it to me as I've picked up the implementation of this issue.

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Verified for site which is in gathering state.
- **Note** : Settings for Force Analytics data under Tester plugin is not working. I tested this for new site which is in gathering state and for the site which SC data is in zero state and Analytics is in gathering state.

@techanvil 

**Issue >** When GA4 and/or SC are in gathering state then On Dashboard ` Key metrics  Setup CTA banner ` is not showing but under settings Key Metrics Admin settings panel is still appearing.

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/a5d1ac76-0b59-421a-924e-d9fd8055d6e3)

![image](https://github.com/google/site-kit-wp/assets/94359491/53b6cf89-20f8-441e-93e7-df6b08e2b6d6)

https://github.com/google/site-kit-wp/assets/94359491/fe2a0647-6b02-4b9e-9881-36819600977b

</details>

**Comment by techanvil** [comment]
Hey @mohitwp, thanks for raising this.

> * **Note** : Settings for Force Analytics data under Tester plugin is not working. I tested this for new site which is in gathering state and for the site which SC data is in zero state and Analytics is in gathering state.

Unfortunately, the short answer here is that the Tester plugin doesn't yet support forcing report data for GA4, so we currently still need to test scenarios relating to the GA4 zero/gathering data states manually.

> **Issue >** When GA4 and/or SC are in gathering state then On Dashboard `Key metrics Setup CTA banner` is not showing but under settings Key Metrics Admin settings panel is still appearing.

Good spot here - I'm raising a followup PR to fix this.

**Comment by aaemnnosttv** [comment]
Back to you for another pass @mohitwp 👍

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Verified KeyMetrics banner/widget not showing when SC and/or GA4 are in gathering state on Main Dashboard and under Admin settings.
- The issue reported [above](https://github.com/google/site-kit-wp/issues/6607#issuecomment-1637998153) is now resolved.
- Tested on site which is in gathering state.
- Tested on site which is in zero data state.
- Tested on site which is not in gathering and zero state.
- Also, tested on site where GA4 is in gathering state and SC is in non-gathering state.


<details>


https://github.com/google/site-kit-wp/assets/94359491/511e1fea-2973-433b-abce-ff3ac78c839a

https://github.com/google/site-kit-wp/assets/94359491/aa1a5370-4056-4f8a-8342-f521d10ac364

https://github.com/google/site-kit-wp/assets/94359491/1ad8e63b-ee50-4f79-9f14-64405791d9e9

</details>
